### PR TITLE
Improve inbox cancellations

### PR DESCRIPTION
**Pull request by alan**
- Fix the date to show the actual cancellation date
- Show the reference number of the booking when expanding the cancellation
- Show which visitors are banned or not on the list

### CODE REVIEW

File: app/views/prison/dashboards/_request_details_row.html.erb [hunk 1]
```diff
@@ -1,24 +1,17 @@
 <tr class="hidden-row hide" aria-hidden="true" aria-expanded="false">
   <td colspan="9" class="request-details">
     <div class="request-row-details">
-      <% if visit.slot_granted? -%>
+      <% visit.slots.each_with_index do |slot, i| %>
         <dl>
-          <dt><%= t('.booked_date') %></dt>
-          <dd><%= format_visit_slot_date_for_staff(visit) %><br><%= format_visit_slot_times_for_staff(visit) %></dd>
+          <dt><%= t('.choice', number: ActiveSupport::Inflector.ordinalize(i+1)) %></dt>
```
Comment (StupidCodeFactory) [inline app/views/prison/dashboards/_request_details_row.html.erb]: probably can do the following: 
```ruby
(i+1).ordinalize
```

File: app/views/prison/dashboards/_request_details_row.html.erb [hunk 2]
```diff
@@ -1,24 +1,17 @@
 <tr class="hidden-row hide" aria-hidden="true" aria-expanded="false">
   <td colspan="9" class="request-details">
     <div class="request-row-details">
-      <% if visit.slot_granted? -%>
+      <% visit.slots.each_with_index do |slot, i| %>
```
Comment (StupidCodeFactory) [inline app/views/prison/dashboards/_request_details_row.html.erb]: What about collection rendering? Then shoving the inside of the loop in a partial:
```ruby
<%= render visit.slots %>
```

File: app/models/visit.rb [hunk 3]
```diff
@@ -164,6 +164,10 @@ def last_visit_state
     visit_state_changes.order('created_at desc').first
   end
 
+  def additional_visitors
+    visitors.reject { |v| v == principal_visitor }
```
Comment (StupidCodeFactory) [inline app/models/visit.rb]: i'd memoize this but it's fine as it is

File: app/views/prison/dashboards/_request_details_row.html.erb (hunk 2 above)
Comment (alan) [inline app/views/prison/dashboards/_request_details_row.html.erb]: `ConcreteSlot` is not an ActiveModel so doesn't implement `#to_partial_path` for this to work. I'm tempted to leave it as it is as the partial won't be reused anywhere else. What do you think?

File: app/views/prison/dashboards/_request_details_row.html.erb (hunk 1 above)
Comment (alan) [inline app/views/prison/dashboards/_request_details_row.html.erb]: I've updated it in the last commit

File: app/models/visit.rb (hunk 3 above)
Comment (alan) [inline app/models/visit.rb]: Good spot, I've updated it

**Pull request merged**